### DISCUSSION TITLE
Unable to decompress zstd backups/exports outside RavenDB

### DISCUSSION THREAD

**Discussion by krivulcik** [ravendb/ravendb#18045] [Q&A] [answered]
I can't decompress backup/export files using Linux zstd.

I get the following error:
```
$ unzstd Dump\ of\ zstd-test\ 2024-01-26\ 11-44.ravendbdump.json.zst
ravendbdump.json.zst : 0 MB...     ravendbdump.json.zst : Read error (39) : premature end
```

The file is a manual export of an empty database (on http://live-test.ravendb.net/, which currently runs 6.0.4-nightly-20240118-0201), renamed to add the `.json.zst` extension.

The same error occurs with backup files.

This happens both on Ubuntu in WSL and native Ubuntu.

Also, Windows Explorer seems to be willing to understand the .zst extension and shows a single file upon entering the compressed file. The file inside is empty, though. But it works only for some export files, and also doesn't work for known correct files, so Windows likely doesn't work well with zstd.

Is it possible to extract raw JSON from the backup/export files on Linux/Windows using generic tools?

**Comment by ppekrol** [comment]
@krivulcik 

If you would want to decompress files, the best way would be to export them and set Gzip compression, this can be done in advanced options in the Studio.

**Reply by krivulcik** [reply]
Yes, that's what I've been doing, but I have a set of zstd exports from an older state (it didn't occur to me to change the compression then), and it would be helpful to be able to decompress them to inspect the exact data in the backups. I'm investigating an issue and can't access the old database state, I have only the exports.

**Reply by ppekrol** [reply]
I think it would be best to use ZstdStream.Decompress from https://github.com/ravendb/ravendb/blob/v6.0/src/Sparrow/Utils/ZstdStream.cs
It is internal, you can either copy it to your app or clone our repository and use it in Tryouts project that we have there.

**Reply by krivulcik** [reply]
Thanks, I'll try that.

However, is it intentional that the files can't be decompressed using other tools?

**Reply by ayende** [reply]
No, it isn't.
I assume that this is because we are storing the raw stream and missing some header

**Comment by ayende** [comment, marked as answer]
This is most probably fixed here: https://github.com/ravendb/ravendb/pull/22314
